cherrypy/process/wspbus.py

```diff
@@ -336,14 +336,15 @@ def block(self, interval=0.1):
         # See https://github.com/cherrypy/cherrypy/issues/751.
         self.log('Waiting for child threads to terminate...')
         for t in threading.enumerate():
-            # Validate the we're not trying to join the MainThread
-            # that will cause a deadlock and the case exist when
+            # Validate the we're not trying to join the MainThread or
+            # DummyThread that will cause a deadlock and the case exist when
             # implemented as a windows service and in any other case
             # that another thread executes cherrypy.engine.exit()
             if (
                     t != threading.currentThread() and
-                    t.isAlive() and
-                    not isinstance(t, threading._MainThread)
+                    not isinstance(t, (threading._MainThread,
+                                       threading._DummyThread)) and
+                    t.isAlive()
```

Comment:
According to [#1342](https://github.com/cherrypy/cherrypy/issues/1342#issuecomment-215957718) *already returns **only alive*** threads for Python 2.6+ (I checked the refs in that comment). Could you please wipe `t.isAlive()` off this condition?